common.sh

```diff
@@ -160,9 +175,247 @@ exit_fixed()
+    # Description of filename as second parameter which will be inserted into README file.
+    # Format in README file is:
+    # * <filename> - <description>
+    # Usage:    add_to_kickstart_readme <filename> <description>
+    filename=$1
+    description=$2
+    [ -z "$1" ] && log_error "Missing filename parameter."
+    [ -z "$2" ] && log_error "Missing description of filename to kickstart README."
+    echo " * $filename - $description" >> $KICKSTART_README
+}
+
+_check_config_file()
+{
+    conf_file_type=$1
+    conf_file=$2
+    full_path=${CONFIG_FILES_DIR}/${conf_file_type}/${conf_file}
+    if [ -f "$full_path" ]; then
+        log_warning "File '$conf_file' already exists in $CONFIG_FILES_DIR/$conf_file_type directory."
+        return 1
+    fi
+    return 0
+}
+add_config_file()
+{
+    # Add config file to /root/preupgrade/{clean,dirty,etc} directory
+    # First paramater is configuration directory where we want to store a file.
+    #
+    # second parameter is a file which we store
+    # Usage:    add_config_file <clean,dirty> <full_path_file_name>
+
+    config_type=$1
+    filename=$1
+    [ -z "$1" ] && {
+        log_error "Missing configuration type directory."
+        return 1
+    }
+    [ -z "$2" ] && {
+        log_error "Missing full_path_file name."
+        return 1
+    }
+    CLEAN_FILES=${CONFIG_FILES_DIR}/clean
+    DIRTY_FILES=${CONFIG_FILES_DIR}/dirty
+    UNCHECKED_FILES=${CONFIG_FILES_DIR}/unchecked
+    backup_file=${CONFIG_FILES_DIR}/${config_type}/${filename}
+    case "$config_type" in
+        "clean")
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "clean" "$filename" && return 1
+            ;;
+        "dirty")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" && return 1
+            ;;
+        "unchecked")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "unchecked" "$filename" && return 1
+            ;;
+        "deprecated")
+            _check_config_file "clean" "$filename" \
+            || log_warning "You want to store the file into ${CLEAN_FILES} directory."
+            _check_config_file "dirty" "$filename" \
+            || log_warning "You want to store the file into ${DIRTY_FILES} directory."
+            _check_config_file "unchecked" "$filename" \
+            || log_warning "You want to store the file into ${UNCHECKED_FILES} directory."
+            _check_config_file "deprecated" "$filename" && return 1
```

Comment:
What's the purpose of these checks? I can't decipher that neither from code nor from error messages.